src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs

```diff
@@ -141,6 +141,40 @@ internal bool IsExceptionType(TypeSymbol type, ref HashSet<DiagnosticInfo> useSi
             return IsEqualOrDerivedFromWellKnownClass(type, WellKnownType.System_Exception, ref useSiteDiagnostics);
         }
 
+        internal bool IsWellKnownTupleType(TypeSymbol typeToCheck, int requiredNumberOfElements)
+        {
+            Debug.Assert(!typeToCheck.IsTupleType, "The helper is meant to be used to validate underlying types, not with actual tuples");
+
+            // NOTE: error type symbol is NamedTypeSymbol, 
+            //       but not SymbolKind.NamedType, so check the kind before casting.
+            if (typeToCheck.Kind != SymbolKind.NamedType)
+            {
+                return false;
+            }
+
+            NamedTypeSymbol currentType = (NamedTypeSymbol)typeToCheck;
+            const int maxNumberOfRegularElements = TupleTypeSymbol.RestPosition;
```

Comment:
The name of this cons is confusing. Its value is 8, but the maximum number of regular elements is 7. Consider making its value 7 by subtracting 1 from the current initializer and adjusting the code appropriately.